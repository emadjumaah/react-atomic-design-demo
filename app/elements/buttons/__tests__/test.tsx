import React from 'react';
import {
  PrimaryButton,
  SecondaryButton,
  TertiaryButton,
  ButtonColor,
} from 'elements';
import { render, fireEvent, cleanup } from '@testing-library/react';

describe('Buttons', () => {
  const onClick = jest.fn();

  beforeEach(() => {
    cleanup();
    onClick.mockClear();
  });

  describe('Primary', () => {
    it('Handles onClick', () => {
      const { getByTestId } = render(
        <PrimaryButton onClick={onClick}>Primary Button</PrimaryButton>,
      );

      fireEvent.click(getByTestId('button'));
      expect(onClick).toHaveBeenCalledTimes(1);
    });

    it('Renders All Variants', () => {
      const { container } = render(
        <>
          <PrimaryButton color={ButtonColor.BLUE} onClick={onClick}>
            Primary
          </PrimaryButton>
          <PrimaryButton color={ButtonColor.ORANGE} onClick={onClick}>
            Primary
          </PrimaryButton>
          <PrimaryButton color={ButtonColor.GREEN} onClick={onClick}>
            Primary
          </PrimaryButton>
          <PrimaryButton color={ButtonColor.BLACK} onClick={onClick}>
            Primary
          </PrimaryButton>
          <PrimaryButton color={ButtonColor.WHITE} onClick={onClick}>
            Primary
          </PrimaryButton>
        </>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Secondary', () => {
    it('Handles onClick', () => {
      const { getByTestId } = render(
        <SecondaryButton onClick={onClick}>Secondary Button</SecondaryButton>,
      );

      fireEvent.click(getByTestId('button'));
      expect(onClick).toHaveBeenCalledTimes(1);
    });

    it('Renders All Variants', () => {
      const { container } = render(
        <>
          <SecondaryButton color={ButtonColor.BLUE} onClick={onClick}>
            Secondary
          </SecondaryButton>
          <SecondaryButton color={ButtonColor.ORANGE} onClick={onClick}>
            Secondary
          </SecondaryButton>
          <SecondaryButton color={ButtonColor.GREEN} onClick={onClick}>
            Secondary
          </SecondaryButton>
          <SecondaryButton color={ButtonColor.BLACK} onClick={onClick}>
            Secondary
          </SecondaryButton>
          <SecondaryButton color={ButtonColor.WHITE} onClick={onClick}>
            Secondary
          </SecondaryButton>
        </>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Tertiary', () => {
    it('Handles onClick', () => {
      const { getByTestId } = render(
        <TertiaryButton onClick={onClick}>Tertiary Button</TertiaryButton>,
      );

      fireEvent.click(getByTestId('button'));
      expect(onClick).toHaveBeenCalledTimes(1);
    });

    it('Renders All Variants', () => {
      const { container } = render(
        <>
          <TertiaryButton color={ButtonColor.BLUE} onClick={onClick}>
            Tertiary
          </TertiaryButton>
          <TertiaryButton color={ButtonColor.ORANGE} onClick={onClick}>
            Tertiary
          </TertiaryButton>
          <TertiaryButton color={ButtonColor.GREEN} onClick={onClick}>
            Tertiary
          </TertiaryButton>
          <TertiaryButton color={ButtonColor.BLACK} onClick={onClick}>
            Tertiary
          </TertiaryButton>
          <TertiaryButton color={ButtonColor.WHITE} onClick={onClick}>
            Tertiary
          </TertiaryButton>
        </>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
