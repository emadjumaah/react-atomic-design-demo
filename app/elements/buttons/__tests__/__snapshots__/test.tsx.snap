// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Primary Renders All Variants 1`] = `
.c0 {
  background-color: var(--primary);
  border: none;
  color: var(--white);
}

.c0:hover {
  cursor: pointer;
}

.c1 {
  background-color: var(--secondary);
  border: none;
  color: var(--white);
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  background-color: var(--tertiary);
  border: none;
  color: var(--white);
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  background-color: var(--black);
  border: none;
  color: var(--white);
}

.c3:hover {
  cursor: pointer;
}

.c4 {
  background-color: var(--white);
  border: none;
  color: var(--black);
}

.c4:hover {
  cursor: pointer;
}

<div>
  <button
    class="c0"
    data-testid="button"
  >
    Primary
  </button>
  <button
    class="c1"
    data-testid="button"
  >
    Primary
  </button>
  <button
    class="c2"
    data-testid="button"
  >
    Primary
  </button>
  <button
    class="c3"
    data-testid="button"
  >
    Primary
  </button>
  <button
    class="c4"
    data-testid="button"
  >
    Primary
  </button>
</div>
`;

exports[`Buttons Secondary Renders All Variants 1`] = `
.c0 {
  background-color: transparent;
  border: 2px solid var(--primary);
  color: var(--primary);
}

.c0:hover {
  cursor: pointer;
}

.c1 {
  background-color: transparent;
  border: 2px solid var(--secondary);
  color: var(--secondary);
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  background-color: transparent;
  border: 2px solid var(--tertiary);
  color: var(--tertiary);
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  background-color: transparent;
  border: 2px solid var(--black);
  color: var(--black);
}

.c3:hover {
  cursor: pointer;
}

.c4 {
  background-color: transparent;
  border: 2px solid var(--white);
  color: var(--white);
}

.c4:hover {
  cursor: pointer;
}

<div>
  <button
    class="c0"
    data-testid="button"
  >
    Secondary
  </button>
  <button
    class="c1"
    data-testid="button"
  >
    Secondary
  </button>
  <button
    class="c2"
    data-testid="button"
  >
    Secondary
  </button>
  <button
    class="c3"
    data-testid="button"
  >
    Secondary
  </button>
  <button
    class="c4"
    data-testid="button"
  >
    Secondary
  </button>
</div>
`;

exports[`Buttons Tertiary Renders All Variants 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  color: var(--primary);
}

.c0:hover {
  cursor: pointer;
}

.c1 {
  background-color: transparent;
  border: none;
  color: var(--secondary);
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  background-color: transparent;
  border: none;
  color: var(--tertiary);
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  background-color: transparent;
  border: none;
  color: var(--black);
}

.c3:hover {
  cursor: pointer;
}

.c4 {
  background-color: transparent;
  border: none;
  color: var(--white);
}

.c4:hover {
  cursor: pointer;
}

<div>
  <button
    class="c0"
    data-testid="button"
  >
    Tertiary
  </button>
  <button
    class="c1"
    data-testid="button"
  >
    Tertiary
  </button>
  <button
    class="c2"
    data-testid="button"
  >
    Tertiary
  </button>
  <button
    class="c3"
    data-testid="button"
  >
    Tertiary
  </button>
  <button
    class="c4"
    data-testid="button"
  >
    Tertiary
  </button>
</div>
`;
